Markdown formwidget improvements

Klaasie opened the pull request:
This PR adds readonly and disabled support to the markdown editor formwidget.
Additionally it adds support for setting html attributes through the form fields yaml.

_The .less file for the markdown editor was not included in the asset compiler. I've added it so that it's in line with other formwidgets._

LukeTowers commented:
@Klaasie could you add before / after screenshots of this change?

Klaasie commented:
@LukeTowers I've made no changes to the default state of the editor. I've modified the javascript to disable click events on the toolbar and set the ace editor to readonly. Added a `disabled` class that counters the hover states and added a `cursor: not-allowed` to several classes to give the visual feedback that this editor can't be clicked.

(if you'd still like a before and after screenshot I can make one later when I'm back on my macbook. Windows won't allow me to make screenshots with the cursor visible.)

LukeTowers commented:
Screenshot would be helpful @Klaasie, thanks!

Klaasie commented:
Before:
![Schermafbeelding 2020-03-26 om 09 01 00](https://user-images.githubusercontent.com/3026259/77623477-746b4a80-6f40-11ea-8285-8f63a463d298.png)
![Schermafbeelding 2020-03-26 om 09 01 10](https://user-images.githubusercontent.com/3026259/77623484-76350e00-6f40-11ea-8c19-4883acacfdfe.png)

You can see, the cursor is a text icon and the toolbar buttons have hover and tooltip states.

After:
![Schermafbeelding 2020-03-26 om 08 57 52](https://user-images.githubusercontent.com/3026259/77623544-9238af80-6f40-11ea-8fad-9f03ef6cf798.png)
![Schermafbeelding 2020-03-26 om 08 58 03](https://user-images.githubusercontent.com/3026259/77623550-949b0980-6f40-11ea-980a-510fd609f030.png)

The cursor shows that this field is disabled, hovers are not showing and tooltips are disabled.

LukeTowers commented:
Can we just hide the toolbar entirely when it's disabled / readonly?

Klaasie commented:
@LukeTowers Yeah can do. See the screenshot below how that'd look.

![image](https://user-images.githubusercontent.com/3026259/77673499-88d43500-6f8a-11ea-802b-4fb7581b0e03.png)

Would you prefer this over blocking any actions within the toolbar through javascript?

LukeTowers commented:
hmm, I almost prefer just rendering the content by itself. What does the richeditor look like in disabled / readonly mode?

Klaasie commented:
![image](https://user-images.githubusercontent.com/3026259/77733766-c925c880-7007-11ea-9990-38a90bb3c40c.png)

This is what the richeditor looks like disabled. Hover effects on the toolbar still work but clicking them does nothing. So does clicking on the content. The cursor does not change, so there's no visual feedback there.

LukeTowers commented:
@Klaasie thanks for looking into that. Can we go with the original suggestion of showing the buttons but just make the colours grey and more muted like the richeditor example?

Klaasie commented:
@LukeTowers I've modified the colors to be the same as the richeditor.
There is still two differences compared to the richeditor:
1. I've counteracted hover states so you don't get the visual feedback that the button can be clicked.
2. I've kept the `cursor: not-allowed` styling just like browsers will do by default on disabled fields.

I personally find those 2 differences make sense from a "disabled" point of view. Once I get to looking into the richeditor I can look into making that toolbar work the same as the markdown toolbar does now.

Additionally I muted the text color as well and I've removed the cursor line indicator that you could still see in my screenshots 2 days ago.

Attached are 2 screenshots of my latest changes.

![markdowneditor_empty_disabled](https://user-images.githubusercontent.com/3026259/77821405-56d9e480-70ea-11ea-900b-b43dd0e6f957.png)
![markdowneditor_filled_disabled](https://user-images.githubusercontent.com/3026259/77821407-58a3a800-70ea-11ea-8684-d6e7503fe475.png)

LukeTowers commented:
@Klaasie I would prefer the content section does not get muted, but I like the toolbar changes (and the cursor in the content field so long as you actually can't type anything into it)
